generator/internal/parser/pagination.go

```diff
@@ -37,8 +38,19 @@ func updateMethodPagination(a *api.API) {
 		var hasPageSize bool
 		var hasPageToken *api.Field
 		for _, f := range reqMsg.Fields {
-			if f.JSONName == pageSize && f.Typez == api.INT32_TYPE {
-				hasPageSize = true
+			// For some legacy services that predates AIP-4233 such as SQL,
+			// maxResults is used instead of pageSize for the field name.
+			// In some cases, the type is also not standardized so we must
+			// support [INT32, UINT32] types.
```

Comment:
"The passive voice shall never be used" and other nits:

 ```suggestion
			// Some legacy services (e.g. sqladmin.googleapis.com)
			// predate AIP-4233 and use `maxResults` instead of 
			// `pageSize` for the field name.
			// Furthermore, some of these services use both 
			// `uint32` and `int32` for the `maxResults` field type.
```